src/APIM_ARMTemplate/apimtemplate/Commands/Extract.cs

```diff
@@ -12,17 +13,170 @@ public ExtractCommand()
             this.Name = Constants.ExtractName;
             this.Description = Constants.ExtractDescription;
 
-            var apimname = this.Option("--name <apimname>", "API Management name", CommandOptionType.SingleValue).IsRequired();
+            var apiManagementName = this.Option("--name <apimname>", "API Management name", CommandOptionType.SingleValue);
+            var resourceGroupName = this.Option("--resourceGroup <resourceGroup>", "Resource Group name", CommandOptionType.SingleValue);
+            var fileFolderName = this.Option("--fileFolder <filefolder>", "ARM Template files folder", CommandOptionType.SingleValue);
+
             this.HelpOption();
 
-            this.OnExecute(() =>
+            this.OnExecute(async () =>
             {
-                if (apimname.HasValue())
-                    Console.WriteLine($"Create command executed with name {apimname.Value()}");
-                else
-                    ColoredConsole.Error.WriteLine("API Management name passed in");
-                return 0;
+                if (!apiManagementName.HasValue()) throw new Exception("Missing parameter <apimname>.");
```

Comment:
One thing we should discuss is whether we should use our manual validation here vs. simply making these fields required with this.Option().isRequired()